code/client/managedsoftwareupdate

```diff
@@ -218,6 +218,24 @@ def runScript(script, display_name, runtype):
         display.display_warning(unicode_or_str(err))
     return result
 
+def cleanup(runtype):
+    """
+    If there are executables inside the cleanup directory, run them and remove them if successful
+    """
+    cleanupdir = os.path.join(os.path.realpath(os.path.dirname(sys.argv[0])), 'cleanup')
+    if not os.path.isdir(cleanupdir):
+        # path doesn't exist or isn't a directory
+        return
+
+    # get all of the files in the cleanup directory
+    (_, _, filenames) = next(os.walk(cleanupdir))
```

Comment:
what's the advantage of this over the simpler `filenames = os.listdir(cleanupdir)` ?

Reply:
None, it's just what I'm used to using. Not married to either if you'd like it to change.

Reply:
Yeah, since I'm usually the one on the hook for maintaining the code after it gets merged, I'd rather it be the simpler version so I'm not confused when I look at it in a year. :-)

Reply:
Oh, the difference is `(_, _, filenames) = next(os.walk(cleanupdir))` only produces a list of files, `filenames = os.listdir(cleanupdir)` is files and directories.

Reply:
OK. Keep it as is, but add a comment for future me:
`# get only filenames (and not directory names) from cleanupdir`